toolchain/check/testdata/interop/cpp/function_decl.carbon

```diff
@@ -45,6 +45,151 @@ fn F() {
   Cpp.foo_float();
 }
 
+// --- single_int_param_function_decl.h
+
+auto foo(int b) -> void;
+
+// --- import_single_int_param_function_decl.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "single_int_param_function_decl.h";
+
+fn F() {
+  Cpp.foo(1);
+}
+
+// --- multiple_int_params_function_decl.h
+
+auto foo(int a, int b) -> void;
+
+// --- import_multiple_int_params_function_decl.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "multiple_int_params_function_decl.h";
+
+fn F() {
+  Cpp.foo(1, 2);
+}
+
+// --- multiple_functions_with_int_params.h
+
+auto foo1(int a, int b) -> void;
+
+auto foo2(int c, int b) -> void;
+
+// --- import_multiple_functions_with_int_params.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "multiple_functions_with_int_params.h";
+
+fn F() {
+  Cpp.foo1(1, 2);
+  Cpp.foo2(3, 4);
+}
+
+// --- unnamed_int_param_function_decl.h
+
+auto foo(int) -> void;
+
+// --- import_unnamed_int_param_function_decl.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "unnamed_int_param_function_decl.h";
+
+fn F() {
+  Cpp.foo(1);
+}
+
+// --- int_ref_param_function_decl.h
+
+auto foo(int& a) -> int;
+
+// --- fail_int_ref_param_function_decl.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "int_ref_param_function_decl.h";
+
+fn F() {
+  var a : i32 = 1;
```

Comment:
Style-wise, we generally aren't putting a space before the `:` in cases like this.
```suggestion
  var a: i32 = 1;
```